ui/hooks/metamask-notifications/useProfileSyncing.test.tsx

```diff
@@ -79,4 +82,88 @@ describe('useProfileSyncing', () => {
 
     expect(actions.disableProfileSyncing).toHaveBeenCalled();
   });
+
+  it('should dispatch account syncing when conditions are met', async () => {
+    store = mockStore({
+      metamask: {
+        isProfileSyncingEnabled: true,
+        isSignedIn: true,
+        isUnlocked: true,
+        useExternalServices: true,
+        participateInMetaMetrics: false,
+        internalAccounts: {
+          accounts: {
+            '0x123': {
+              address: '0x123',
+              id: 'account1',
+              metadata: {},
+              options: {},
+              methods: [],
+              type: 'eip155:eoa',
+            },
+          },
+        },
+      },
+    });
+
+    store.dispatch = jest.fn().mockImplementation((action) => {
+      if (typeof action === 'function') {
+        return action(store.dispatch, store.getState);
+      }
+      return Promise.resolve();
+    });
+
+    jest.clearAllMocks();
```

Comment:
Since these tests use similar mocks, can we refactor and create an `arrangeMocks` function?

(long term) I kind of want to clean up this file too.
- Rather than 1 big describe, lets create multiple describe blocks for the separate tests
- Refactor & Reused the store and dispatch mocks.
- (minor) Not a fan of beforeEach, it's always felt flakey. I prefer arrangeMock fns. But IIWIW.